### PR TITLE
[POC] Add basic file operations

### PR DESCRIPTION
**Pull request by micbar**
Needs #8 first

```gherkin
Feature: Basic File operations
  As a user
  I want to be able to do basic file operations
  So that I can organize my files

  Background: Space Exists
    Given user "marie" has logged in with password "radioactivity"
    Given user "marie" has created a personal space with the alias "Maries Home"

    Scenario: Create Folders and files
      Given user "marie" has created a folder "NewFolder" in the home directory with the alias "NewFolder"
      And user "marie" has uploaded a file "testfile.txt" with content "lorem ipsum" in the home directory with the alias "testfile"
      When user "marie" lists all resources inside the resource with alias "Maries Home"
      Then the following resources should be listed in the response:
        | type      | path         |
        | container | NewFolder    |
        | file      | testfile.txt |

    Scenario: Create Tree
      Given user "marie" has created a folder "A" in the home directory with the alias "A"
      And user "marie" has created a folder "A/B" in the home directory with the alias "B"
      And user "marie" has uploaded a file "A/B/testfile.txt" with content "lorem ipsum" in the home directory with the alias "testfile"
      When user "marie" lists all resources inside the resource with alias "Maries Home"
      Then the following resources should not be listed in the response:
        | type      | path         |
        | container | B            |
        | file      | testfile.txt |
      And the following resources should be listed in the response:
        | type      | path |
        | container | A    |

  Scenario: Rename resources
    Given user "marie" has created a folder "A" in the home directory with the alias "A"
    And user "marie" has uploaded a file "testfile.txt" with content "lorem ipsum" in the home directory with the alias "testfile"
    When user "marie" lists all resources inside the resource with alias "Maries Home"
    Then the following resources should be listed in the response:
      | type      | path         |
      | container | A            |
      | file      | testfile.txt |
    When user "marie" has moved the resource with alias "A" inside a space to target "A-Folder"
    And user "marie" has moved the resource with alias "testfile" inside a space to target "Test-File.txt"
    When user "marie" lists all resources inside the resource with alias "Maries Home"
    Then the following resources should be listed in the response:
      | type      | path          |
      | container | A-Folder      |
      | file      | Test-File.txt |
    And the following resources should not be listed in the response:
      | type      | path         |
      | container | A            |
      | file      | testfile.txt |
```

### CODE REVIEW

File: features/file-operations.feature [hunk 1]
```diff
@@ -0,0 +1,50 @@
+Feature: Basic File operations
+  As a user
+  I want to be able to do basic file operations
+  So that I can organize my files
+
+  Background: Space Exists
+    Given user "marie" has logged in with password "radioactivity"
+    Given user "marie" has created a personal space with the alias "Maries Home"
```
Comment (phil-davis) [inline features/file-operations.feature]:
```suggestion
    And user "marie" has created a personal space with the alias "Maries Home"
```

File: features/file-operations.feature [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    Given user "marie" has created a folder "A" in the home directory with the alias "A"
+    And user "marie" has uploaded a file "testfile.txt" with content "lorem ipsum" in the home directory with the alias "testfile"
+    When user "marie" lists all resources inside the resource with alias "Maries Home"
+    Then the following resources should be listed in the response:
+      | type      | path         |
+      | container | A            |
+      | file      | testfile.txt |
+    When user "marie" has moved the resource with alias "A" inside a space to target "A-Folder"
+    And user "marie" has moved the resource with alias "testfile" inside a space to target "Test-File.txt"
+    When user "marie" lists all resources inside the resource with alias "Maries Home"
```
Comment (phil-davis) [inline features/file-operations.feature]:
```suggestion
    When user "marie" moves the resource with alias "A" inside a space to target "A-Folder"
    And user "marie" moves the resource with alias "testfile" inside a space to target "Test-File.txt"
    And user "marie" lists all resources inside the resource with alias "Maries Home"
```

In `When` steps we usually write the step text in active present tense.

File: steps/resources/steps.go [hunk 3]
```diff
@@ -272,6 +274,57 @@ func (f *ResourcesFeatureContext) ResourceOfTypeShouldBeListedInTheResponse(numb
...
+	list, ok := f.Response.(*providerv1beta1.ListContainerResponse)
+	if !ok {
+		return fmt.Errorf("expected to receive a ListContainerResponse but got something different")
+	}
+	rows := table.Rows
+	if len(rows) == 0 {
+		return fmt.Errorf("empty gherkin table")
+	}
+	if rows[0].Cells[0].Value != "type" && rows[0].Cells[1].Value != "path" {
+		return fmt.Errorf("the first line of the tables needs to be in the form <type><path>")
```
Comment (individual-it) [inline steps/resources/steps.go]:
```suggestion
		return fmt.Errorf("the first line of the tables needs to be in the form | <type> | <path> |")
```

**Review by individual-it — COMMENTED** [review]
just a minor change

File: features/file-operations.feature (hunk 1 above)
Comment (micbar) [reply features/file-operations.feature:8]: please also change the step implementation

File: features/file-operations.feature (hunk 2 above)
Comment (micbar) [reply features/file-operations.feature]: please also change the step implementation

**Review by phil-davis — APPROVED** [review]

File: features/file-operations.feature (hunk 2 above)
Comment (phil-davis) [reply features/file-operations.feature]: done

**Pull request merged**